File: javaScript/problem2.js
// Considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
function sumOfFibonacci() {
  var limit = 4000000
  var a = 0;
  var b = 1;
  var sum = 0;

  while (a < limit) {
    if ( a % 2 === 0) {
      sum += a;
    }
    var c = a + b;
    a = b;
    b = c;
  }
  return sum;
}

console.log(sumOfFibonacci());

File: rubyChallenge/problem2.rb
# project euler problem2 in rb

def sum_of_fibonacci
  limit = 4_000_000
  a = 0
  b = 1
  sum = 0

  while a < limit
    sum += a if a % 2 == 0

    c = a + b
    a = b
    b = c
  end
  sum
end

p sum_of_fibonacci
